twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/query-builder.ts

```diff
@@ -49,13 +49,16 @@ export function buildSelectQuery<Entity>(
     })
     .filter(Boolean);
 
-  const query = `SELECT * FROM ${options.keyspace}.${entityDefinition.name} WHERE ${[
+  let query = `SELECT * FROM ${options.keyspace}.${entityDefinition.name} WHERE ${[
     ...where,
     ...(buildComparison(findOptions) || []),
     ...(buildIn(findOptions) || []),
     ...(buildLike(findOptions) || []),
-  ].join(" AND ")}`;
+  ].join(" AND ")}`.trimEnd();
 
+  if (query.endsWith(" WHERE")) {
+    query = query.slice(0, -" WHERE".length)
+  }
   return query;
```

Comment:
Why don't you create a variable where before and then decide to append it or not ?

```
const where = [ ...where,
    ...(buildComparison(findOptions) || []),	    ...(buildComparison(findOptions) || []),
    ...(buildIn(findOptions) || []),	    ...(buildIn(findOptions) || []),
    ...(buildLike(findOptions) || []),	    ...(buildLike(findOptions) || []),
  ].join(" AND ")}`;

const query = "SELECT * FROM ${options.keyspace}.${entityDefinition.name} " + (where.length ? (" WHERE "+where) : "")
```

Reply:
Why not :)